script.js

```diff
@@ -0,0 +1,368 @@
+    div.dataset.code = this.code;
+    div.innerHTML = this.code;
+    return div;
+  }
+}
+
+function creatButton() {
+  const ar = [];
+  const arr = Object.keys(keyboardElements);
+  arr.forEach((el) => {
+    ar.push(
+      new Button({
+        code: el,
+        key: keyboardElements[el][0],
+        style: keyboardElements[el][0],
+      })
+    );
+  });
+  ar.forEach((el) => {
+    keyboard.append(el.buildButton());
+  });
+}
+
+creatButton();
+showKey();
+textarea.focus();
+
+function showKey() {
+  let arrValues = Object.values(keyboardElements);
+
+  arrValues.forEach((element) => {
+    if (element[0] === 'rShift') {
+      document.querySelector('.rShift').after(clearfix);
+    }
+  });
+
+  if (shiftCase === false && sessionStorage.lang === 'ENG') {
+    arrValues.forEach((element) => {
+      let a = document.querySelector(`.${element[0]}`);
+      if (element.length < 2) {
+        a.innerText = element[0];
+      } else if (element[0] === 'enter') {
+      } else if (element[1] !== undefined) {
+        a.innerText = element[1];
+      }
+    });
+  }
+
+  if (shiftCase === true && sessionStorage.lang === 'ENG') {
+    arrValues.forEach((element) => {
+      let a = document.querySelector(`.${element[0]}`);
+      if (element.length < 2) {
+        a.innerText = element[0];
+      } else if (element[0] === 'enter') {
+      } else if (element[2] !== undefined) {
+        a.innerText = element[2];
+      }
+    });
+  }
+
+  if (shiftCase === false && sessionStorage.lang === 'RUS') {
+    arrValues.forEach((element) => {
+      let a = document.querySelector(`.${element[0]}`);
+      if (element.length < 2) {
+        a.innerText = element[0];
+      } else if (element[0] === 'enter') {
+      } else if (element[3] !== undefined) {
+        a.innerText = element[3];
+      }
+    });
+  }
+
+  if (shiftCase === true && sessionStorage.lang === 'RUS') {
+    arrValues.forEach((element) => {
+      let a = document.querySelector(`.${element[0]}`);
+      if (element.length < 2) {
+        a.innerText = element[0];
+      } else if (element[0] === 'enter') {
+      } else if (element[4] !== undefined) {
+        a.innerText = element[4];
+      }
+    });
+  }
+}
+
+keyboard.addEventListener('mousedown', function (event) {
+  textarea.focus();
+  if (event.target.classList[0] == 'keyboard') return;
+  lastPressButton = event.target;
+  setStyle(event.target, true);
+  identifyKey(event.target.dataset, 'down');
+  showKey();
+});
+
+keyboard.addEventListener('mouseup', function (event) {
+  textarea.focus();
+  if (event.target.dataset.code === 'CapsLock') return;
+  removeLastPressButtonStyle();
+  identifyKey(event.target.dataset, 'up');
+  showKey();
+});
+
+document.addEventListener('keydown', function (event) {
+  if (event.repeat) return;
+  event.preventDefault();
+  setStyle(document.querySelector(`.${keyboardElements[event.code][0]}`), true);
+  identifyKey(event, 'down');
+  checkLanguage();
+  showKey();
```

Comment:
Why do we need to `showKey` on every keydown? Moreover `checkLanguage` can call `showKey` also just before `showKey`.  So there are some situations when `showKey` can be called twice.